tests/sources/test_github.py

```diff
@@ -863,6 +866,34 @@ async def test_validate_config_missing_fields_then_raise(field):
             await source.validate_config()
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "host_url,should_raise",
+    [
+        pytest.param("http://github.example.com", True, id="HTTP URL should be rejected"),
+        pytest.param("https://github.example.com", False, id="HTTPS URL should be accepted"),
+    ]
+)
+async def test_validate_config_https_url_validation(host_url, should_raise):
```

Comment:
The other tests in this test module mock methods like `get_personal_access_token_scopes` - but that method seems like an implementation detail (maybe it's even public just to allow mocking which seems like a code smell).

Instead of doing the same here I mock a response from the remote server - I guess this is a bit less brittle to refactoring but also assumes things about the connector internals. 

Whoever is reviewing, if you have an opinion on doing things one or the other way, let me know.

Reply:
I like using `aioresponses` here